lib/buildkite/test_collector/ci.rb

```diff
@@ -79,7 +82,8 @@ def circleci
       "url" => ENV["CIRCLE_BUILD_URL"],
       "branch" => ENV["CIRCLE_BRANCH"],
       "commit_sha" => ENV["CIRCLE_SHA1"],
-      "number" => ENV["CIRCLE_BUILD_NUM"],
+      "build_number" => ENV["CIRCLE_BUILD_NUM"],
+      "build_id" => ENV["CIRCLE_WORKFLOW_ID"]
```

Comment:
I think I may have got my understanding of workflows wrong when I set the default value for `key` on circle CI.

I think you're correct in choosing `CIRCLE_WORKFLOW_ID` as synonymous with `BUILDKITE_BUILD_ID`. We may need to fix line 81 to jus use `CIRCLE_WORKFLOW_ID` 🤔 What do you think?

If I'm understanding the relationship of circle workflows and builds correctly, I think `CIRCLE_BUILD_NUMBER` is closer to `BUILDKITE_JOB_NUMBER` (if it existed) than it is to `BUILDKITE_BUILD_NUMBER`.

Maybe we can chat about this one in breakout?

Reply:
Ok I've had a hoon around the circleci docs, and also asked ChatGPT and it looks like circleci doesn't have an equivelent of `BUILDKITE_BUILD_NUMBER` or `GITHUB_RUN_ATTEMPT`. ChatGPT says you could use `CIRCLE_BUILD_NUM` with custom logic to work it out, but I think this might be lies given it gave me a totally wrong answer the first time I asked.
I *think* CircleCI creates a whole new workflow id on retry given this [answer on the community pages](https://discuss.circleci.com/t/detect-rerun-of-a-job/44735), so `WORKFLOW_ID` might be good enough?

Reply:
I checked with an old colleague who still uses circleci and she did some digging around and pretty much validated the community pages answer. `CIRCLE_WORKFLOW_WORKSPACE_ID` does not change on rerun, but `CIRCLE_WORKFLOW_ID` does, so we can use these in tandem to create the equivalent of a BUILD_ID.

example data:
For the 0th parallel run in the **RSpec** workflow for that CI run, those are:
```
  CIRCLE_WORKFLOW_ID=aaaa... <--- match
  CIRCLE_WORKFLOW_JOB_ID=bbbb...
  CIRCLE_WORKFLOW_WORKSPACE_ID=aaaa... <--- match
```
For the 1st parallel run in the **RSPec** workflow for that CI run, those are the same:
```
  CIRCLE_WORKFLOW_ID=aaaa...
  CIRCLE_WORKFLOW_JOB_ID=bbbb...
  CIRCLE_WORKFLOW_WORKSPACE_ID=aaaa...
```
For the **Rubocop** workflow for that CI run, those are:
```
  CIRCLE_WORKFLOW_ID=aaaa...
  CIRCLE_WORKFLOW_JOB_ID=cccc...
  CIRCLE_WORKFLOW_WORKSPACE_ID=aaaa...
```
If you **re-run** CI and look at the **Rubocop** workflow, they’re:
```
  CIRCLE_WORKFLOW_ID=eeee...
  CIRCLE_WORKFLOW_JOB_ID=dddd...
  CIRCLE_WORKFLOW_WORKSPACE_ID=aaaa...
```

Reply:
Is `CIRCLE_WORKFLOW_ID` unique across all the things or only within a workspace? ie can we use it on its own?

Reply:
Argh I see I got build_number and build_id confused here; yes `CIRCLE_WORKFLOW_ID` is unique, but circleCi doesn't have a concept of an incrementer variable like buildkite (BUILDKITE_BUILD_NUMBER) and github does (GITHUB_RUN_NUMBER). 
On rerun, the `CIRCLE_WORKFLOW_ID` changes, so without the CIRCLE_WORKFLOW_WORKSPACE_ID you can't tell the difference between a new workflow and a rerun.

Do we care about this? Do we set build_number and build_id to the same value here?